components/script/dom/bindings/cell.rs

```diff
@@ -12,14 +12,15 @@ use style::thread_state::{self, ThreadState};
 /// This extends the API of `std::cell::RefCell` to allow unsafe access in
 /// certain situations, with dynamic checking in debug builds.
 #[derive(Clone, Debug, Default, MallocSizeOf, PartialEq)]
-pub struct DomRefCell<T> {
-    value: RefCell<T>,
+//#[must_root] // TODO should it be #[must_root]? then -> there are usages with #[must_root] types and with normal ones; two different structs for them? or hacking plugin compiler further to mark structure as #[must_root] if it generic type is #[must_root]?
```

Comment:
This is an interesting question. What we want to represent is that `DomRefCell<T>` is `must_root` if and only if `T` is `must_root` for each particular instance of T. Is this something that is possible with the current plugin?

Reply:
I see the intention. It is what I came up with, but didn't implement so far. After short recalling I am not sure if it was very problematic in some way. It could potentially require some changes to part of the plugin that checks only HIR (`check_struct_def` method if I remember the name correctly) and maybe some cooperation with the method that analyzes the MIR.